.chloggen/return_error_for_obsreport_new_functions.yaml

```diff
@@ -0,0 +1,18 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'deprecation'
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: obsreport
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "`obsreport.New[Receiver|Scraper|Processor|Exporter]` returns error now, deprecate `obsreport.MustNew` methods"
```

Comment:
We need 2 "chloggen" entries (that's what we do in this case):
1. deprecation -> Must*
2. breaking -> change signature of already deprecated